src/coreclr/src/tools/Common/System/EnumExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using ILCompiler.DependencyAnalysis.ReadyToRun;
+using Internal.JitInterface;
+using Internal.ReadyToRunConstants;
+
+namespace System
+{
+    /// <summary>
+    /// Enum.CompareTo takes an Object, so using that requires boxing the enum we are comparing to (which is slow
```

Comment:
I do not think it makes sense to have these in the shared file.

Instead, I would just add casts to int to every place where this is used (like what I have suggested below).